File: server/model/contract.py
from beanie import Document
from pydantic import BaseModel, Field
from typing import List, Optional
from uuid import uuid4, UUID


class Clause(BaseModel):
    text: str


class Subsection(BaseModel):
    title: str
    clauses: List[Clause] = []


class SectionRaw(BaseModel):
    title: str
    text: str


class Section(BaseModel):
    title: str
    clauses: List[Clause] = []
    subsections: List[Subsection] = []


class Party(BaseModel):
    name: str


class ContractRaw(BaseModel):
    parties: List[Party] = []
    title: str
    sections: List[SectionRaw] = []


class Contract(Document):
    id: Optional[UUID] = Field(default_factory=uuid4, alias="_id")
    title: str
    parties: List[Party] = []
    sections: List[Section] = []

File: client/src/types/contract.ts
/**
 * TypeScript interfaces mirroring the backend contract model
 */

export interface Clause {
  text: string;
}

export interface Subsection {
  title: string;
  clauses: Clause[];
}

export interface SectionRaw {
  title: string;
  text: string;
}

export interface Section {
  title: string;
  clauses: Clause[];
  subsections: Subsection[];
}

export interface Party {
  name: string;
}

export interface ContractRaw {
  parties: Party[];
  title: string;
  sections: SectionRaw[];
}

export interface Contract {
  title: string;
  parties: Party[];
  sections: Section[];
}
